packages/api/src/task/scheduler.ts

```diff
@@ -347,54 +347,50 @@ export class TaskScheduler {
   async createAndScheduleTask(
     type: Task["type"],
     params: Task["params"],
+    user: User,
     inputAsset?: Asset,
     outputAsset?: Asset,
     userId?: string,
     requesterId?: string,
   ) {
+    const projectId = inputAsset?.projectId || user?.defaultProjectId;
     const task = await this.createTask(
       type,
       params,
+      userId,
+      projectId,
       inputAsset,
       outputAsset,
-      userId,
       requesterId,
     );
 
-    let uId = inputAsset?.userId || outputAsset?.userId || userId;
-
-    if (uId) {
-      const user = await db.user.get(uId);
-
-      if (user?.disabled) {
-        throw new Error("user is disabled");
-      }
-    }
-
     await this.scheduleTask(task);
     return task;
   }
 
   async createTask(
     type: Task["type"],
     params: Task["params"],
+    userId?: string,
+    projectId?: string,
     inputAsset?: Asset,
     outputAsset?: Asset,
-    userId?: string,
     requesterId?: string,
   ) {
+    const uId = inputAsset?.userId || outputAsset?.userId || userId;
```

Comment:
`userId` arg should be supreme, no need to grab it from other places. In fact, if any of these ids mismatch it should be a fatal error (no need to check tho, I think we can just assume they are the same)